### PR TITLE
For #24524 - Refactor FaviconListItem into SyncedTabs

### LINKED ISSUE #24524: Replace list item in SyncedTabs with new ListItem composable

### PR DESCRIPTION
**Pull request by MozillaNoah**
Closes #24524 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt [hunk 1]
```diff
@@ -186,53 +184,6 @@ fun SyncedTabsSectionHeader(
...
-                text = tabTitleText,
-                modifier = Modifier.fillMaxWidth(),
-                fontSize = 16.sp,
-                maxLines = 1,
-            )
-
-            SecondaryText(
-                text = url,
-                modifier = Modifier.fillMaxWidth(),
-                fontSize = 12.sp,
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:229]: This is 14.sp in the `ListItem`. Can we double check with what is desired in the designs?

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt [hunk 2]
```diff
@@ -344,10 +295,12 @@ private fun SyncedTabsListItemsPreview() {
...
 
-            SyncedTabsTabItem(
-                tabTitleText = "Mozilla",
-                url = "www.mozilla.org"
-            ) { println("Clicked tab") }
+            FaviconListItem(
+                label = "Mozilla",
+                description = "www.mozilla.org",
+                url = "www.mozilla.org",
+                onClick = { },
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:302]:
```suggestion
                onClick = {},
```

File: app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt (hunk 1 above)
Comment (MozillaNoah) [reply app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt:229]: I was in the middle of writing a comment 😅. You're too quick for me, @gabrielluong

**Comment by MozillaNoah** [comment]
@cwzilla, with this change, the font size for the description/URL increased from 12 to 14. The original spec for Synced Tabs used 12, but the current List Item component uses 14 for the description. Is this increase intended, or should the List Item component be decreased to 12?

Before:
image


After:
Screen Shot 2022-04-21 at 3 10 07 PM

**Comment by cwzilla** [comment]
@MozillaNoah This was intended to improve the accessibility :) I think the default material list items use size 14 in the description too.

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**